pkg/metrics/wal/wal_test.go

```diff
@@ -372,6 +372,24 @@ func TestStorage_TruncateAfterClose(t *testing.T) {
 	require.Error(t, ErrWALClosed, s.Truncate(0))
 }
 
+func BenchmarkAppendExemplar(b *testing.B) {
+	walDir, _ := ioutil.TempDir(os.TempDir(), "wal")
+	defer os.RemoveAll(walDir)
+
+	s, _ := NewStorage(log.NewNopLogger(), nil, walDir)
+	app := s.Appender(context.Background())
+	sRef, _ := app.Append(0, labels.Labels{{Name: "a", Value: "1"}}, 0, 0)
+
+	e := exemplar.Exemplar{Labels: labels.Labels{{Name: "a", Value: "1"}}, Value: 20, Ts: 10, HasTs: true}
```

Comment:
Hmm, I think this is going to inflate the results since every exemplar is always the same. (i.e., with the fix, you're always doing label comparisons, which is going to be slower). 

I'm not sure how dramatic the impact would be though, so it might not be worth changing to generate some new + existing exemplars every time.

Reply:
After changing to use exemplars.Equal(), I checked the difference between changing the label, the timestamp and the value inside the benchmark loop. Indeed, there's quite some difference there, related to which field we compare first/last.

One thought is that the inflated result may be okay since (at least regarding exemplar comparison) we're getting an upper bound? We could also have three exemplars and modulo choose a different one each time.

```bash
# Comparing Label values in benchmark loop
# e.Label[0].Value = strconv.Itoa(i)
# Extra allocation due to Itoa conversion
name              time/op        alloc/op        allocs/op
AppendExemplar-8  97.6ns ± 4%     56.0B ± 0%       2.00 ± 0%

# Comparing exemplar timestamp
# e.Ts = int64(i)
name              time/op        alloc/op        allocs/op
AppendExemplar-8  144ns ± 2%     335B ± 2%       1.00 ± 0%

# Comparing exemplar values in benchmark loop
# e.Value = float64(i)
name              time/op        alloc/op        allocs/op
AppendExemplar-8  146ns ± 5%     340B ± 1%       1.00 ± 0%
```

Reply:
Hmm, I think this is probably fine as-is and we can return to it if appending exemplars turns out to be a cpu/memory bottleneck.